conda/core/index.py

```diff
@@ -145,12 +145,12 @@ def _supplement_index_with_cache(index):
             index[pcrec] = pcrec
 
 
-def _make_virtual_package(name, version=None, build_string='0'):
+def _make_virtual_package(name, version=None, build_string=None):
     return PackageRecord(
             package_type=PackageType.VIRTUAL_SYSTEM,
             name=name,
             version=version or '0',
-            build_string=build_string,
+            build_string=(build_string if build_string is not None else '0'),
```

Comment:
```suggestion
            build_string=build_string or '0',
```

Reply:
Oh, I misremembered that these are allowed to be empty.
Looking at the code again, I think you are right, and they should always be non-empty.